openedx/features/caliper_tracking/transformers/problem_transformers.py

```diff
@@ -194,60 +193,33 @@ def problem_check(current_event, caliper_event):
     :param caliper_event: caliper_event log having some basic attributes.
     :return: updated caliper_event.
     """
-    if current_event.get('event_source') == 'browser':
-        return problem_check_browser(current_event, caliper_event)
-    else:
-        return problem_check_server(current_event, caliper_event)
 
-
-def problem_check_browser(current_event, caliper_event):
     caliper_event.update({
         'action': 'Submitted',
         'type': 'AssessmentEvent',
         'object': {
             'id': current_event['referer'],
             'type': 'Assessment',
-            'extensions': {
-                'event': current_event['event']
-            }
         }
     })
     caliper_event['referrer']['type'] = 'WebPage'
     caliper_event['actor'].update({
         'name': current_event['username'],
         'type': 'Person'
     })
-    caliper_event['extensions']['extra_fields'].update({
-        'course_id': current_event['context']['course_id'],
-        'ip': current_event['ip']
-    })
-    return caliper_event
-
+    caliper_event['extensions']['extra_fields'].update(current_event['context'])
+    caliper_event['extensions']['extra_fields']['ip'] = current_event['ip']
 
-def problem_check_server(current_event, caliper_event):
```

Comment:
Why these changes?

Reply:
Because we can use a single method for three events. Previously there were two methods to a single difference in two event logs. Now both can be transformed using this transformer and also this one.

Reply:
Alright.